src/ProofTools/DoubleCutDeletionTool.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * Deletes legal double cuts
+ * @author Dawn Moore
+ */
+
+import {Point} from "../AEG/Point";
+import {AtomNode} from "../AEG/AtomNode";
+import {CutNode} from "../AEG/CutNode";
+import {drawCut, redrawTree} from "../DrawModes/DrawUtils";
+import {tree} from "../index";
+import {illegalColor} from "../Themes";
+import {offset} from "../DrawModes/DragMode";
+
+//The node selected with the user mouse down.
+let currentNode: CutNode | AtomNode | null = null;
+
+//Whether or not the node is allowed to be moved (not the sheet).
+let legalNode: boolean;
+
+/**
+ * Takes the current point and finds the lowest node containing that point.
+ * If that node is a double cut then it is a legal node and highlights it with the illegal color.
+ * @param event The event of a mouse down while the user is using double cut deletion
+ */
+export function doubleCutDeletionMouseDown(event: MouseEvent) {
+    const currentPoint: Point = new Point(event.x - offset.x, event.y - offset.y);
+    currentNode = tree.getLowestNode(currentPoint);
+
+    if (currentNode instanceof CutNode && isDoubleCut(currentNode)) {
+        legalNode = true;
+        highlightDoubleCut(currentNode);
+    } else {
+        legalNode = false;
+    }
+}
+
+/**
+ * Selects a new current node and if that node is a legal double cut highlights the double cut.
+ * Removes any previous highlighting.
+ * @param event The event of a mouse move while the user is using double cut deletion
+ */
+export function doubleCutDeletionMouseMove(event: MouseEvent) {
+    const currentPoint: Point = new Point(event.x - offset.x, event.y - offset.y);
+    currentNode = tree.getLowestNode(currentPoint);
+    redrawTree(tree);
+
+    if (currentNode instanceof CutNode && isDoubleCut(currentNode)) {
+        legalNode = true;
+        highlightDoubleCut(currentNode);
+    } else {
+        legalNode = false;
+    }

```

Comment:
Since you use this twice, i recommend factoring it out into its own helper function.